src/lx/parser.act

```diff
@@ -350,17 +370,20 @@
 
 	<compile-regex>: (s :string, f :flags) -> (r :fsm) = @{
 		struct re_err err;
-		const char *s;
+		struct re_buf buf;
```

Comment:
Thanks for this comment. I was thinking about it, but we don't need the offset bit for this and these live in automatic storage anyway. Also, string is used by things that aren't RE_NATIVE/RE_LITERAL regexps, and so that felt a little gross. I think in general the allocation overhead for the `struct string` will dwarf the copy overhead into automatic storage here.

So that's why I didn't do it.